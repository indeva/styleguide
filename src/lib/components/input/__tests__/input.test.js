import React from 'react';
import renderer from 'react-test-renderer';
import Input from '..';

describe('Input', () => {
  it('renders the input', () => {
    const component = renderer.create(<Input label="Test" value="" onChange={() => {}}>Test</Input>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the lined input', () => {
    const component = renderer.create(<Input label="Test" value="" onChange={() => {}} lined>Test</Input>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the input as password', () => {
    const component = renderer.create(<Input label="Test" value="" onChange={() => {}} type="password">Test</Input>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
