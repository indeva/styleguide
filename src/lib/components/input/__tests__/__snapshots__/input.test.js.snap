// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders the input 1`] = `
<input
  className="input"
  onChange={[Function]}
  type="text"
  value=""
/>
`;

exports[`Input renders the input as password 1`] = `
<input
  className="input"
  onChange={[Function]}
  type="password"
  value=""
/>
`;

exports[`Input renders the lined input 1`] = `
<input
  className="input input--lined"
  onChange={[Function]}
  type="text"
  value=""
/>
`;
